mpf/devices/logic_blocks.py

```diff
@@ -304,6 +304,87 @@ def _initialize(self):
             self.hit_value *= -1
         elif self.config['direction'] == 'up' and self.hit_value < 0:
             self.hit_value *= -1
+        # Add control events if included in the config
+        if self.config['control_events']:
+            self._setup_control_events(self.config['control_events'])
+
+    def add_control_events_in_mode(self, mode: Mode) -> None:
+        """Do not auto enable this device in modes."""
+        pass
+
+    def _setup_control_events(self, event_list):
+        self.debug_log("Setting up control events")
+
+        kwargs = {}
+        for entry in event_list:
+            if entry['action'] in ('add', 'subtract', 'set_value'):
+                handler = getattr(self, entry['action'])
+                kwargs = {'value': entry['value']}
+            else:
+                raise AssertionError("Invalid control_event action {} in mode".
+                                     format(entry['action']), self.name)
+            self.machine.events.add_handler(entry['event'], handler, **kwargs)
+
+    def check_complete(self, count_complete_value=None):
+        '''
+        Returns true if the counter has reached or surpassed its specified
+        completion value, returns False if no completion criteria or is
+        not complete.
+        '''
+        # If count_complete_value was not passed, obtain it
+        if count_complete_value is None:
+            count_complete_value = self.config['count_complete_value']\
+                .evaluate([]) if self.config['count_complete_value']\
+                is not None else None
+
+        if count_complete_value is not None:
+
+            if self.config['direction'] == 'up' and self._state.value >= count_complete_value:
+                return True
+
+            elif self.config['direction'] == 'down' and self._state.value <=\
+                    count_complete_value:
+                return True
+        return False
+
+    def add(self, **kwargs):
```

Comment:
See @jabdoa2 's recent work on event handlers to consider separating the handler into method `self.event_add` that accepts kwargs and calls `self.add` which does _not_ accept kwargs. Same for the other handler methods.